searx/redislib.py

```diff
@@ -0,0 +1,201 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""A collection of convenient functions and redis/lua scripts.
+"""
+# https://redis.com/blog/bullet-proofing-lua-scripts-in-redispy/
+
+import hmac
+
+from searx import get_setting
+
+PURGE_BY_PREFIX = """
+local prefix = tostring(ARGV[1])
+for i, name in ipairs(redis.call('KEYS', prefix .. '*')) do
+    redis.call('EXPIRE', name, 0)
```

Comment:
EXPIRE removes the object asynchronous while DEL is executed synchronous.

Reply:
I couldn't find information about `DEL` vs `EXPIRE 0`.

According to [this micro benchmark](https://gist.github.com/dalf/d43a06ad4e1c5c3913ff57b3278da22a#file-redis_expire_vs_del-py),  `EXPIRE` is more or less equivalent to `DEL` is the best case scenario, otherwise `DEL` is faster especially when there are a lot of keys. May be, I have this point. Could you document the choice of EXPIRE instead of DEL?

Reply:
Its not about what command is faster .. its about transaction frames in the DB itself .. if there is an open transaction frame in the DB the `DEL` has to wait until the frame is closed, then `DEL` can delete the key .. when you use `EXPIRE` the call of the `EXPIRE` turns back immediately and the python command loop can continue .. the deletion (expire) of the key in the DB is managed by the DB internally.

In line 5 of `searx/redislib.py` I gave a link to [Bullet-Proofing Lua Scripts in RedisPy](https://redis.com/blog/bullet-proofing-lua-scripts-in-redispy/) .. its a very interesting article, may you like to read it.